linera-indexer/src/graphql.rs

```diff
@@ -0,0 +1,98 @@
+};
+use linera_chain::data_types::{
+    notifications::Notification, Event, ExecutedBlock, HashedValue, IncomingMessage, Origin,
+    OutgoingMessage,
+};
+use linera_execution::{committee::Epoch, Message, Operation};
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../linera-explorer/graphql/schema.graphql",
+    query_path = "../linera-explorer/graphql/notifications.graphql",
+    response_derives = "Debug"
+)]
+pub struct Notifications;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../linera-explorer/graphql/schema.graphql",
+    query_path = "../linera-explorer/graphql/block.graphql",
+    response_derives = "Debug"
+)]
+pub struct Block;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../linera-explorer/graphql/schema.graphql",
+    query_path = "../linera-explorer/graphql/chains.graphql",
+    response_derives = "Debug"
+)]
+pub struct Chains;
+
+impl From<block::BlockBlockValueExecutedBlockBlockIncomingMessages> for IncomingMessage {
+    fn from(val: block::BlockBlockValueExecutedBlockBlockIncomingMessages) -> Self {
+        IncomingMessage {
+            origin: val.origin,
+            event: val.event,
+        }
+    }
+}
+
+impl From<block::BlockBlockValueExecutedBlockBlock> for linera_chain::data_types::Block {
+    fn from(val: block::BlockBlockValueExecutedBlockBlock) -> Self {
+        let incoming_messages: Vec<IncomingMessage> = val
+            .incoming_messages
+            .into_iter()
+            .map(|b: block::BlockBlockValueExecutedBlockBlockIncomingMessages| b.into())
+            .collect();
+        linera_chain::data_types::Block {
+            chain_id: val.chain_id,
+            epoch: val.epoch,
+            incoming_messages,
+            operations: val.operations,
+            height: val.height,
+            timestamp: val.timestamp,
+            authenticated_signer: val.authenticated_signer,
+            previous_block_hash: val.previous_block_hash,
+        }
+    }
+}
+
+impl From<block::BlockBlockValueExecutedBlockMessages> for OutgoingMessage {
+    fn from(val: block::BlockBlockValueExecutedBlockMessages) -> Self {
+        OutgoingMessage {
+            destination: val.destination,
+            authenticated_signer: val.authenticated_signer,
+            message: val.message,
+        }
+    }
+}
+
+impl From<block::BlockBlockValueExecutedBlock> for ExecutedBlock {
+    fn from(val: block::BlockBlockValueExecutedBlock) -> Self {
+        let messages: Vec<OutgoingMessage> = val
+            .messages
+            .into_iter()
+            .map(|b: block::BlockBlockValueExecutedBlockMessages| b.into())
+            .collect();
+        ExecutedBlock {
+            block: val.block.into(),
+            messages,
+            state_hash: val.state_hash,
+        }
+    }
+}
+
+impl From<block::BlockBlock> for HashedValue {
+    fn from(val: block::BlockBlock) -> Self {
+        HashedValue::new_confirmed(val.value.executed_block.into())
```

Comment:
Do we know that this is always a `ConfirmedBlock`?

Reply:
I thought the notifications were only done for confirmed block, but I probably misunderstood

Reply:
No, you're right.
But it's still a bit weird conceptually to implement `From` this way for the type itself: It's not an inherent property of the type that it's always a confirmed block, just a property of the context it's being used in.
Doesn't `block::BlockBlock` know which variant it is? Then this `From` implementation could be done correctly, even if only the `ConfirmedBlock` case is relevant for now.

Reply:
yes it is possible, I just missed the `CertificateValue` step, I can fix this.